### DISCUSSION TITLE
[Quadlet] Separating caddy ang php-fpm

### DISCUSSION THREAD

**Discussion by rEnr3n** [containers/podman#22230] [Q&A]
I'm trying to migrate wallabag to quadlet. Before quadlet, I had 5 systemd services.

1. Pod
2. Wallabag
I manually replaced `ExecStart` to use `podman create`. `Type=oneshot` and `RemainAfterExit=yes` were added too. My Containerfile looked something like this:
```
FROM docker.io/archlinux AS build
RUN pacman --quiet --noconfirm --needed -Syu wallabag

FROM scratch
COPY --from=build /usr/share/wallabag /
```
3. Postgresql
4. Caddy
Caddy needs access to wallabag files so they are mounted like this:
`--mount type=image,src=wallabag,dst=/usr/share/wallabag,rw=true`
`--volumes-from wallabag` for the bind mounts
5. Php-fpm
Php also need access to the files so they're also mounted here:
`--mount type=image,src=wallabag,dst=/usr/share/wallabag,rw=true`
`--volumes-from wallabag` for the bind mounts

---

Now my dilemma is running `wallabag` container. I don't see any option in quadlet that allows me to use `podman create` instead of `podman run`. The container doesn't have a `sleep` command I can use to keep it running.

So I tried to merge wallabag and php-fpm containers into one but realized caddy needs access to the files too. I would need to install caddy, php-fpm and wallabag into one container. I didn't like that idea. I tried to create a systemd drop-in file like this:
```
[Service]
ExecStart=
ExecStart=podman create ...
```
But t doesn't remove the `ExecStart=podman run ...` line.

I'v ran out of ideas on how to proceed with this. Any help would be appreciated.

**Comment by ygalblum** [comment]
Hi,

I'm trying to understand the use case. IIUC the sole purpose of the wallabag service is building the image. Am I correct?

If yes, what I think you could possibly do is manually write this specific service (as you did) and set it as a dependency for the PHP and Caddy ones.

I did have a vision of expanding the `.image` files to also build images (and not only pull them). But, I didn't get around to implementing it.

**Reply by rEnr3n** [reply]
> If yes, what I think you could possibly do is manually write this specific service (as you did) and set it as a dependency for the PHP and Caddy ones.

That was my setup. The problem is the image contains only files that comprise the entire wallabag webapp. There is no `sleep` or `php` command there so I can't keep it running.

My current workaround is to download the webapp on my host and mount it on php-fpm and caddy containers. I'm not sure if I like this route since I'm still figuring out how to automate this.
